packages/merge/src/components/MergeTable/Selection.js

```diff
@@ -71,22 +71,54 @@ export const RelationsCheckBoxes = React.memo(({
   setSelectedMultiple,
   selectedMultiple
 }) => {
-  return entityData.value.map(relationEntity => {
-    const isSelected = _get(selectedMultiple, [name, relationEntity.key]) === entityKey
-    const onChange = e => setSelectedMultiple(name, entityKey, relationEntity.key, e.target.checked)
+  const isSelected = key => _get(selectedMultiple, [name, key]) === entityKey
+  const setSelected = (key, checked) => setSelectedMultiple(name, entityKey, key, checked)
 
-    return (
-      <StyledLabelWrapper key={`relation-checkbox-${relationEntity.key}`}>
-        <input type="checkbox"
-          onChange={onChange}
-          checked={isSelected}
-          id={`${name}${entityKey}-${relationEntity.key}`}/>
-        <Typography.Label for={`${name}${entityKey}-${relationEntity.key}`}>
-          <FormattedValue type="single-select" value={relationEntity}/>
-        </Typography.Label>
-      </StyledLabelWrapper>
-    )
-  })
+  const numberTotal = entityData.value.length
+  const numberSelected = entityData.value.filter(relationEntity => isSelected(relationEntity.key)).length
+  const allRowsSelectionState = numberTotal === numberSelected
+    ? 'checked'
+    : numberSelected > 0
+      ? 'indeterminate'
+      : 'unchecked'
+  const allRowsSelectionChange = v => {
+    entityData.value.forEach(relationEntity => {
+      const checked = v === 'checked'
+      if (isSelected(relationEntity.key) !== checked) {
+        setSelected(relationEntity.key, checked)
+      }
+    })
+  }
+
+  return <>
+    {
+      entityData.value.length === 0
```

Comment:
` entityData.value.length > 0 && <StyledLabelWrapper....
`
Should be sufficient